ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -345,177 +144,58 @@ public void close() throws IOException {
-            AccountServiceErrorCode.BadRequest);
-      }
-      short accountId = container.getParentAccountId();
-      short containerId = container.getId();
-      return mySqlAccountStore.getDatasetVersion(accountId, containerId, accountName, containerName, datasetName,
-          version);
-    } catch (SQLException e) {
-      throw translateSQLException(e);
-    }
-  }
-
-  @Override
-  public void deleteDatasetVersion(String accountName, String containerName, String datasetName,
-      String version) throws AccountServiceException {
-    try {
-      Container container = getContainerByName(accountName, containerName);
-      if (container == null) {
-        throw new AccountServiceException("Can't find the container: " + containerName + " in account: " + accountName,
-            AccountServiceErrorCode.BadRequest);
-      }
-      short accountId = container.getParentAccountId();
-      short containerId = container.getId();
-      mySqlAccountStore.deleteDatasetVersion(accountId, containerId, datasetName, version);
-    } catch (SQLException e) {
-      throw translateSQLException(e);
-    }
-  }
-
-  @Override
-  public List<DatasetVersionRecord> getAllValidVersion(String accountName, String containerName, String datasetName)
+  public Collection<Container> updateContainers(String accountName, Collection<Container> containers)
       throws AccountServiceException {
-    try {
-      Container container = getContainerByName(accountName, containerName);
-      if (container == null) {
-        throw new AccountServiceException("Can't find the container: " + containerName + " in account: " + accountName,
-            AccountServiceErrorCode.BadRequest);
-      }
-      short accountId = container.getParentAccountId();
-      short containerId = container.getId();
-      return mySqlAccountStore.getAllValidVersion(accountId, containerId, datasetName);
-    } catch (SQLException e) {
-      throw translateSQLException(e);
-    }
-  }
-
-  @Override
-  public List<DatasetVersionRecord> getAllValidVersionsOutOfRetentionCount(String accountName,
-      String containerName, String datasetName) throws AccountServiceException {
-    try {
-      Container container = getContainerByName(accountName, containerName);
-      if (container == null) {
-        throw new AccountServiceException("Can't find the container: " + containerName + " in account: " + accountName,
-            AccountServiceErrorCode.BadRequest);
+    Collection<Container> containerList = cachedAccountService.updateContainers(accountName, containers);
+    if (config.enableNewDatabaseForMigration) {
+      try {
+        cachedAccountServiceNew.updateContainers(accountName, containers);
+      } catch (AccountServiceException e) {
+        // if before migration, update container for new db would fail due to the account does not exist.
+        logger.trace("This is expected due to the account {} does not exist in new db", accountName);
```

Comment:
We will firstly enable due write then do the one time dump. But I can still add the metrics since later it will be insync for most of the cases unless something is wrong.